scripts/utils/termux_reuse_pr_build_artifacts.sh

```diff
@@ -0,0 +1,233 @@
+  || infoexit "PR head does not match pushed commit changes, probably PR ref was force pushed right after PR was merged. Not performing CI fast path."
+
+  # obtain the common ancestor commit where the PR diverged from master.
+  PR_MERGE_BASE="$(git merge-base "ref/tmp/$PR_BASE_COMMIT" "$HEAD_COMMIT")" || infoexit "failed to obtain PR merge base, not performing CI fast path"
+
+  # obtain list of all files changed in master branch since this PR diverged
+  PR_BASE_TO_HEAD_CHANGED_FILES="$(git diff-tree --name-only -r "$PR_MERGE_BASE..$OLD_COMMIT")" ||:
+
+  # obtain list of all packages changed by this PR
+  PR_CHANGED_PACKAGES="$(
+    git diff-tree --name-only -r "$OLD_COMMIT..$HEAD_COMMIT" \
+      | grep -E "^($DIRS_REGEX)/[^/]+/" \
+      | sed -E "s#^(($DIRS_REGEX)/[^/]+)/.*#\1#" \
+      | sort -u
+  )" ||:
+
+  # obtain list of all buildsystem files changed since this PR diverged
+  PR_BASE_TO_HEAD_CHANGED_BUILDSYSTEM_FILES="$(
+    grep -e "^scripts/" -e "^ndk-patches/" -e "^build-package.sh$" <<< "$PR_BASE_TO_HEAD_CHANGED_FILES"
+  )" ||:
+
+  # obtain list of all packages changes since this PR diverged
+  PR_BASE_TO_HEAD_CHANGED_PACKAGES="$(
+    echo "$PR_BASE_TO_HEAD_CHANGED_FILES" \
+      | grep -E "^($DIRS_REGEX)/[^/]+/" \
+      | sed -E "s#^(($DIRS_REGEX)/[^/]+)/.*#\1#" \
+      | sort -u
+  )" ||:
+
+  # obtain list of all build dependencies of packages changed by this PR
+  PR_CHANGED_PACKAGES_DEPS="$(
+    for i in ${PR_CHANGED_PACKAGES:-}; do
+      ./scripts/buildorder.py $i $TERMUX_PACKAGE_DIRECTORIES | awk '{print $NF}'
+    done | sort -u
+  )" ||:
+
+  # get list of all build dependencies changed since this PR diverged
+  PR_BASE_TO_HEAD_CHANGED_DEPS="$(grep -Fxf <(echo "$PR_BASE_TO_HEAD_CHANGED_PACKAGES") - <<< "$PR_CHANGED_PACKAGES_DEPS")" ||:
+
+  echo "Packages changed by this PR:" ${PR_CHANGED_PACKAGES:-none}
+  echo "Dependencies of these packages changed since PR divergence: " ${PR_BASE_TO_HEAD_CHANGED_DEPS:-}
+  echo "Buildsystem files changed since PR divergence:" ${PR_BASE_TO_HEAD_CHANGED_BUILDSYSTEM_FILES:-none}
+  echo "Packages updated since PR divergence:" ${PR_BASE_TO_HEAD_CHANGED_PACKAGES:-none}
+
+  if [[ -z "${PR_BASE_TO_HEAD_CHANGED_BUILDSYSTEM_FILES:-}${PR_BASE_TO_HEAD_CHANGED_DEPS:-}" ]] || (( PR_CI_REUSE )); then
+    # Same commit can be used in more than one pull request or even push
+    # No hard tabs or newlines allowed inside a graphql request.
+
+    RESPONSE="$(graphql_request "
+    query {
+      repository(owner: \"termux\", name: \"termux-packages\") {
+        object(oid: \"$PR_HEAD_COMMIT\") {
+          ... on Commit { checkSuites(first: 16) { nodes { workflowRun { event file { path } databaseId } conclusion status } } }
+        }
+      }
+    }" "Failed to perform GraphQL request, not performing CI fast path")"
+    # Obtain the first completed and successfull `packages.yml` workflow run.
+    WORKFLOW_ID="$(jq -r '[.data.repository.object?.checkSuites?.nodes[]?
```

Comment:
This should probably also get dropped down a line so it isn't quite as indented.

Reply:
Done.